identity-apps-core/apps/recovery-portal/src/main/webapp/includes/init-url.jsp

```diff
@@ -0,0 +1,169 @@
+            }
+            if (StringUtils.isBlank(userTenantDomain)) {
+                userTenantDomain = tenantDomain;
+            }
+            if (StringUtils.isNotBlank(tenantDomainOfUser)) {
+                tenantForTheming = tenantDomainOfUser;
+            }
+            if (StringUtils.equals(spAppName, "My Account")
+                && StringUtils.equals(tenantDomain, IdentityManagementEndpointConstants.SUPER_TENANT)) {
+                tenantForTheming = userTenantDomain;
+                userTenant = userTenantDomain;
+            }
+        }
+    } else {
+        tenantDomain = request.getParameter(TENANT_DOMAIN_KEY);
+        if (StringUtils.isBlank(tenantDomain) && StringUtils.isNotBlank((String) request.getAttribute(TENANT_DOMAIN_KEY))) {
+            tenantDomain = (String) request.getAttribute(TENANT_DOMAIN_KEY);
+        }
+
+        String tenantDomainFromURL = request.getParameter(TENANT_DOMAIN_SHORT);
+        if (StringUtils.isBlank(tenantDomainFromURL) && StringUtils.isNotBlank((String) request.getAttribute(TENANT_DOMAIN_SHORT))) {
+            tenantDomainFromURL = (String) request.getAttribute(TENANT_DOMAIN_SHORT);
+        }
+
+        String tenantDomainOfUser = request.getParameter(USER_TENANT_DOMAIN_SHORT);
+        if (StringUtils.isBlank(tenantDomainOfUser) && StringUtils.isNotBlank((String) request.getAttribute(USER_TENANT_DOMAIN_SHORT))) {
+            tenantDomainOfUser = (String) request.getAttribute(USER_TENANT_DOMAIN_SHORT);
+        }
+
+        spAppName = request.getParameter(SERVICE_PROVIDER_NAME_SHORT);
+        if (StringUtils.isBlank(spAppName) && StringUtils.isNotBlank((String) request.getAttribute(SERVICE_PROVIDER_NAME_SHORT))) {
+            spAppName = (String) request.getAttribute(SERVICE_PROVIDER_NAME_SHORT);
+        }
+
+        spAppId = request.getParameter(SERVICE_PROVIDER_ID_SHORT);
+        if (StringUtils.isBlank(spAppId) && StringUtils.isNotBlank((String) request.getAttribute(SERVICE_PROVIDER_ID_SHORT))) {
+            spAppId = (String) request.getAttribute(SERVICE_PROVIDER_ID_SHORT);
+        }
+
+        if (StringUtils.isBlank(tenantDomain)) {
+            tenantDomain = request.getParameter(IdentityManagementEndpointConstants.TENANT_DOMAIN);
+            if (StringUtils.isBlank(tenantDomain) && StringUtils.isNotBlank((String) request.getAttribute(IdentityManagementEndpointConstants.TENANT_DOMAIN))) {
+                tenantDomain = (String) request.getAttribute(IdentityManagementEndpointConstants.TENANT_DOMAIN);
+            }
+        }
+
+        if (!StringUtils.isBlank(tenantDomainFromURL)) {
+            tenantDomain = tenantDomainFromURL;
+        }
+
+        tenantForTheming = tenantDomain;
+        userTenant = tenantDomain;
+        userTenantDomain = tenantDomainOfUser;
+
+        if (StringUtils.equals(tenantDomain, IdentityManagementEndpointConstants.SUPER_TENANT)
+            && StringUtils.equals(spAppName, "Console")) {
```

Comment:
Will it be possible to take "Console" through a defined constant?

Reply:
There was no defined constant. Added a new constant in the jsp file itself for clarity of code. 
https://github.com/wso2/identity-apps/pull/5817/commits/877baa435b1c4ba3cfffef80cd1e5a56dc36d0c6